source/com.microsoft.tfs.client.clc/src/com/microsoft/tfs/client/clc/commands/Command.java

```diff
@@ -914,6 +915,58 @@ protected final WorkspaceInfo determineCachedWorkspace(final String[] pathFreeAr
         return determineCachedWorkspace(pathFreeArguments, false);
     }
 
+    protected final WorkspaceInfo determineCachedWorkspace(
+        final String[] pathFreeArguments,
+        final boolean ignoreWorkspaceOptionValue)
+        throws CannotFindWorkspaceException,
+            InvalidOptionValueException,
+            InvalidOptionException {
+
+        /*
+         * On all platforms but Mac OSX, we do not have issues with the computer
+         * names detected by CLC.
+         */
+        if (!Platform.isCurrentPlatform(Platform.MAC_OS_X)) {
+            return determineCachedWorkspaceImpl(pathFreeArguments, ignoreWorkspaceOptionValue);
+        }
+
+        /*
+         * On the Mac OSX we should check two versions of the short computer
+         * name to ensure compatibility with the JetBrains' method of its
+         * compilation. The code should be removed when and if this
+         * compatibility is not needed anymore.
+         */
+        WorkspaceInfo workspaceInfo = null;
+        try {
+            workspaceInfo = determineCachedWorkspaceImpl(pathFreeArguments, ignoreWorkspaceOptionValue);
+
+            if (LocalHost.getShortName().equalsIgnoreCase(workspaceInfo.getComputer())) {
```

Comment:
I don't understand the need for the comparison here. If the workspace was found, shouldn't we just return it?